### DISCUSSION TITLE
I can't create app in budibase, the list is not showing

### DISCUSSION THREAD

**Discussion by seoulaja** [Budibase/budibase#2071] [Help] [answered]
Hi guys,

I installed budibase in my office linux server, I used docker compose for installation, it works perfectly fine. I can create super admin and can acces the screens. But when I try to create app, I have nothing in it. But when I tried to create a new app with same name with previous one, I can't save it, it looks like the app is created but not showing. When I check the developer network in chrome I got this red messages

![image](https://user-images.githubusercontent.com/15101654/126766402-1e592511-9446-4237-9a60-74b5674ba41c.png)

Not sure what happen. This is the console logs

`couchdb-service_1     | [notice] 2021-07-23T09:46:46.431399Z couchdb@127.0.0.1 <0.6329.0> e2139e9ccb couchdb-service:5984 172.20.0.8 budibase GET /app_dev_0e89064baf804760836225667844a723/ 200 ok 5
bbapps                | INFO  [1627033606433] (28 on 1fa7891438f3): request errored
bbapps                |     res: {
bbapps                |       "statusCode": 500,
bbapps                |       "headers": {
bbapps                |         "vary": "Accept-Encoding",
bbapps                |         "content-type": "application/json; charset=utf-8",
bbapps                |         "content-length": "66"
bbapps                |       }
bbapps                |     }
bbapps                |     err: {
bbapps                |       "type": "Error",
bbapps                |       "message": "failed with status code 500",
bbapps                |       "stack":
bbapps                |           Error: failed with status code 500
bbapps                |               at ServerResponse.onResFinished (/app/node_modules/pino-http/logger.js:73:38)
bbapps                |               at ServerResponse.emit (events.js:326:22)
bbapps                |               at ServerResponse.EventEmitter.emit (domain.js:483:12)
bbapps                |               at onFinish (_http_outgoing.js:735:10)
bbapps                |               at onCorkedFinish (_stream_writable.js:673:5)
bbapps                |               at afterWrite (_stream_writable.js:490:5)
bbapps                |               at afterWriteTick (_stream_writable.js:477:10)
bbapps                |               at processTicksAndRejections (internal/process/task_queues.js:83:21)
bbapps                |     }
bbapps                |     responseTime: 125
bbapps                |     req: {
bbapps                |       "id": 33,
bbapps                |       "method": "GET",
bbapps                |       "url": "/api/deployments",
bbapps                |       "headers": {
bbapps                |         "host": "103.109.154.62:10000",
bbapps                |         "x-budibase-app-id": "app_dev_0e89064baf804760836225667844a723",
bbapps                |         "dnt": "1",
bbapps                |         "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36",
bbapps                |         "content-type": "application/json",
bbapps                |         "accept": "*/*",
bbapps                |         "referer": "http://103.109.154.62:10000/builder/app/app_dev_0e89064baf804760836225667844a723",
bbapps                |         "accept-encoding": "gzip, deflate",
bbapps                |         "accept-language": "en-AU,en;q=0.9,id;q=0.8,en-US;q=0.7",
`
I tried to install budibase in my local machine using windows, it's working perfectly.

Can you please help me is there any step that I'm mising, perhaps network permission, etc?

Regards,
Seoul

**Comment by mike12345567** [comment]
Hi @seoulaja - just to check, what architecture is your office linux server? Currently Budibase does not support ARM architectures due to the CouchDB 3.0 image not being ARM compatible.

**Reply by seoulaja** [reply]
Hi @mike12345567 - thanks for quick response. When I did uname -m I got this info: x86_64. So it's not ARM right? Sorry I'm not really expert on linux things, but I know how to use it a bit.

**Reply by mike12345567** [reply]
Thanks, that indeed isn't ARM, looks like that should work, what version of docker/docker-compose do you have installed on the machine?

**Reply by seoulaja** [reply]
When I did docker version I got this version: 18.09.4

and when I check my local machine docker, the version is: 20.10.7

is it possible because I use outdated version?

**Reply by mike12345567** [reply]
yes its possible, also an outdated version of docker-compose could be at fault, not sure, but trying to diagnose, i'm unable to re-create the issue (I run linux myself and just span up a new Budibase cluster which is working fine) so might be worth trying a newer docker/docker-compose version.

Also worth noting on linux docker/docker-compose are versioned/installed separately and they aren't entirely backwards compatible, so if you have a very new version of compose it might not work correctly with an older version of docker.

**Reply by seoulaja** [reply]
Hi @mike12345567 - Let me try to install latest docker and see how it goes. Thanks for very quick response! I'll be back soon!

**Reply by seoulaja** [reply]
Hi @mike12345567 - I tried to remove old docker and install it again now I have docker version 20.10.7 but unfortunately the problem still exists. I recoded what happen here: [video](https://www.loom.com/share/c454e9c873614544a69e823b5df172cf)

and this is a screen capture of the error log 
![image](https://user-images.githubusercontent.com/15101654/126805610-3a3e38b8-dec5-43b3-9fbc-a72612c4d26e.png)

Thanks,

**Reply by mike12345567** [reply]
Hi @seoulaja thanks for trying that, it looks like you have an issue with your instances Redis connection, could you check the state of your Redis instance? A `docker container list` should show you the containers running and then you can look at the logs for redis by performing a `docker logs <redis container id>`.

**Reply by seoulaja** [reply]
Hi @mike12345567 - This is the result

`1:C 23 Jul 2021 15:27:26.883 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 23 Jul 2021 15:27:26.883 # Redis version=6.2.5, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 23 Jul 2021 15:27:26.883 # Configuration loaded
1:M 23 Jul 2021 15:27:26.884 * monotonic clock: POSIX clock_gettime
1:M 23 Jul 2021 15:27:26.886 * Running mode=standalone, port=6379.
1:M 23 Jul 2021 15:27:26.886 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
1:M 23 Jul 2021 15:27:26.886 # Server initialized
1:M 23 Jul 2021 15:27:26.886 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
1:M 23 Jul 2021 15:27:26.887 * Loading RDB produced by version 6.2.4
1:M 23 Jul 2021 15:27:26.887 * RDB age 606 seconds
1:M 23 Jul 2021 15:27:26.887 * RDB memory usage when created 0.81 Mb
1:M 23 Jul 2021 15:27:26.887 * DB loaded from disk: 0.001 seconds
1:M 23 Jul 2021 15:27:26.887 * Ready to accept connections
`

Then I tried to add vm.overcommit_memory=1 as suggested but the problem still exits then I googled and found this [link](https://github.com/redis/redis/issues/6123#issuecomment-512808174) and this is what I did :

1. echo 1024 > /proc/sys/net/core/somaxconn with root
2.  add net.core.somaxconn=65535 in the /etc/sysctl.conf file and restart

Then when I open my budi base I can see 37 applications in my screen (you will see there is an application updated 19 days ago)!

![image](https://user-images.githubusercontent.com/15101654/126811039-dd75cca1-140b-4afb-81db-6f15052c607a.png)


_I once was lost, but now I am found, Was blind, but now I see._

Anyway thanks @mike12345567  for your help, I never thought the problem was in Redis.

Terima kasih (Thank you in Bahasa Indonesia)

**Reply by mike12345567** [reply]
Glad a solution was found! I've actually gone and added some changes so that if a situation like this arises again it'll throw a more Redis related error to suggest the correct path!

**Comment by mike12345567** [comment, marked as answer]
answer found here:
https://github.com/Budibase/budibase/discussions/2071#discussioncomment-1042933
